2022/incoming/st-nicks-reindeers-need-water.pod

```diff
@@ -0,0 +1,152 @@
+on his various I<watch> lists.
+
+Just recently, Santa created this script that uses C<Util::H2O> to
+add accessors to data I<in flight> in order to make it easier to
+manage the response C<HASH> reference returned by L<HTTP::Tiny>.
+
+  use strict;
+  use warnings;
+  use JSON       qw//;
+  use HTTP::Tiny qw//;
+  use Util::H2O; # only exports 'h2o'
+  
+Santa can slip C<h2o> inline to his C<GET> request with more ease than
+he's able to slip into a chimney! Can I<you> spot the addition of C<h2o>?
+
+  # give's Santa "$response->content", "$response->status", "$response->success", etc
+  # from HTTP::Tiny's response object (pure HASH)
+  
+  my $http = HTTP::Tiny->new;
+  my $response = h2o $http->get(q{https://jsonplaceholder.typicode.com/users});
+
+Checking for success is clean and easy,
+
+  # check for unsuccessful web request
+
+  if (not $response->success) {
+  
+    print STDERR qq{Can't get list of online persons to watch!\n};
+    printf STDERR qq{Web request responded with with HTTP status: %d\n}, $response->status;
+    exit 1;
+  
+  }
+
+The JSON C<content> in the I<HTTP> call is also easily accessed so that
+it may be turned into a Perl data structure via C<JSON::decode_json>.
+
+  # decode JSON from response content
+  my $json_array_ref = JSON::decode_json($response->content); # $json is an ARRAY reference
+
+In this case and after referencing the API response specification, Santa
+can see that he gets an C<ARRAY> of C<HASH> references:
+
+  # [
+  #  {
+  #    "id": 1,
+  #    "name": "Leanne Graham",
+  #    "username": "Bret",
+  #    "email": "Sincere@april.biz",
+  #    "address": {
+  #      "street": "Kulas Light",
+  #      "suite": "Apt. 556",
+  #      "city": "Gwenborough",
+  #      "zipcode": "92998-3874",
+  #      "geo": {
+  #        "lat": "-37.3159",
+  #        "lng": "81.1496"
+  #      }
+  #    }
+  #  },
+  # ...
+  # ]
+  
+And even though C<h2o> only works on C<HASH> references, this provides
+I<yet another> opportunity for Santa to use C<h2o> on each I<item> in
+the C<ARRAY>, which makes dealing with each person's record much more
+jolly:
+  
+  print qq{lat, lng, name, Twitter\n};
+  
+  foreach my $person (@$json_array_ref) {
+  
+    # -recurse creates deep accessors, e.g.,
+    #  $person->address->geo->lat;
+    
+    h2o -recurse, $person;
+    
+    printf qq{%5.4f, %5.4f, %s, @%s\n},
+           $person->address->geo->lat,   # deep chain of accessors from '-recurse'
+           $person->address->geo->lng,   # deep chain of accessors from '-recurse'
+           $person->name,
+           $person->username;
+  }
+  
+  # print status to STDERR on succcess
+  printf STDERR qq{\nSummary:\nWeb request responded with with HTTP status: %d\n}, $response->status;
+
+=head1 Conclusion
+
+With St. Nick's list now complete with online usage profiles for all people
+on Twitter, he now sets his mind on more jolly thoughts; not sugar plums and
```

Comment:
See above comment about Twitter. I found it a bit confusing to reference it here when it doesn't occur in the example source data.

Reply:
Thank you, I will try to clarify. It's not super important, just trying to reference the "online profile" thing.